libs/full/runtime_distributed/src/server/runtime_support_server.cpp

```diff
@@ -371,7 +368,6 @@ namespace hpx { namespace components { namespace server {
                 dijkstra_cond_ = std::make_unique<hpx::latch>(2);
```

Comment:
Assuming `dijkstra_termination_detection` is not invoked concurrently (which should be the case, otherwise the correctness of the whole implementation [even with the lock] needs to be reviewed), this `dijkstra_cond_` setting code would not be executed concurrently] with other accesses to `dijkstra_cond_`: This setting code happens before the initial locality starts passing the token, while the only other accesses to `dijkstra_cond_` outside of the function `dijkstra_termination_detection` happens in action `dijkstra_termination` which is invoked after the initial locality finally gets back the token.

Reply:
We will be blocked at `dijkstra_cond_->arrive_and_wait(1)` until the other thread calling `dijkstra_cond_->count_down(1)`. We will only reach `dijkstra_cond_.reset()` after `dijkstra_cond_->arrive_and_wait(1)` returns. `arrive_and_wait` and `count_down` are protected by the latch's own mutex lock. I think it should be okay?

Reply:
I see. The latch is using `notify_one` in the `count_down` method and you worry `count_down` will still be trying to make another call to `notify_one` after `arrive_and_wait` exits?